app/cdap/components/FetchIconsFromWidget/utils.tsx

```diff
@@ -0,0 +1,107 @@
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import {
+  fetchConnectionDetails,
+  fetchConnectors,
+  getCategoriesToConnectorsMap,
+} from 'components/Connections/Create/reducer';
+import DataPrepStore from 'components/DataPrep/store';
+import DataPrepActions from 'components/DataPrep/store/DataPrepActions';
+import {
+  IConnectorArray,
+  IConnectorDetailPayloadArray,
+  IConnectorTypes,
+} from 'components/FetchIconsFromWidget/types';
+import WidgetSVG from 'components/FetchIconsFromWidget/Widget';
+import { ImportDatasetIcon } from 'components/WrangleHome/Components/WrangleCard/iconStore/ImportDatasetIcon';
+import React from 'react';
+
+export const getWidgetData = async () => {
+  const connectorTypes: IConnectorArray[] = await fetchConnectors();
+  const connectorDataArray: IConnectorTypes[] = [];
+  const connectorDataWithSvgArray: IConnectorArray[] = [];
+  const allConnectorsPluginProperties: Map<
+    string,
+    IConnectorDetailPayloadArray[]
+  > = getCategoriesToConnectorsMap(connectorTypes);
+  const connectionPayloadArray: IConnectorDetailPayloadArray[] = [];
+  allConnectorsPluginProperties?.forEach((connectorsArray) => {
+    if (connectorsArray.length) {
+      connectorsArray.map((item) => {
+        connectionPayloadArray.push(item);
+      });
+    }
+  });
+
+  const connectionDetailsData = await Promise.all(
+    connectionPayloadArray.map(async (item, index) => {
+      const selectedConnector = {
+        artifact: item.artifact,
+        category: item.category,
+        name: item.name,
+        type: item.type,
+      };
+      connectorDataArray.push(selectedConnector);
+      return new Promise((resolve, reject) => {
+        const response = fetchConnectionDetails(selectedConnector);
+        if (response) {
+          resolve(response);
+        }
+      });
+    })
+  );
+  const connectorWidgetJson = connectionDetailsData.map(
+    ({ connectorWidgetJSON }) => connectorWidgetJSON
+  );
+
+  connectorDataArray.map((connectorType) => {
+    let connectorTypeHasWidget = false;
+    /**
+     * Getting widget icons for connector types
+     */
+    connectorWidgetJson.map((item) => {
+      if (item['display-name'] && item['display-name'].includes(connectorType.name)) {
+        connectorDataWithSvgArray.push({
+          ...connectorType,
+          SVG: <WidgetSVG dataSrc={item?.icon?.arguments?.data} label={connectorType.name} />,
+        });
+        connectorTypeHasWidget = true;
+      }
+    });
+    /**
+     * Retaining the connector types which are not part of widget api
+     */
+    if (!connectorTypeHasWidget) {
+      connectorDataWithSvgArray.push({
+        ...connectorType,
+        SVG: <WidgetSVG dataSrc={undefined} label={connectorType.name} />,
+      });
+    }
+  });
+
+  connectorDataWithSvgArray.push({
+    name: 'Imported Dataset',
+    SVG: ImportDatasetIcon,
```

Comment:
rename svg to camelCase as it is a vaiable
if it is a component, keep TitleCase

Reply:
done